plutus-core/plutus-core/src/PlutusCore/Constant/Meaning.hs

```diff
@@ -232,94 +230,73 @@ type x ~?~ y = TryUnify (x === y) x y
 -- | Get the element at an @i@th position in a list.
 type Lookup :: forall a. Nat -> [a] -> a
 type family Lookup n xs where
+    Lookup _ '[]       = TypeError ('Text "Not enough elements")
     Lookup 0 (x ': xs) = x
     Lookup n (_ ': xs) = Lookup (n - 1) xs
 
 -- | Get the name at the @i@th position in the list of default names. We could use @a_0@, @a_1@,
 -- @a_2@ etc instead, but @a@, @b@, @c@ etc are nicer.
-type GetName i = Lookup i '["a", "b", "c", "d", "e", "f", "g", "h"]
+type GetName :: GHC.Type -> Nat -> Symbol
+type family GetName k i where
+    GetName GHC.Type i = Lookup i '["a", "b", "c", "d", "e", "i", "j", "k", "l"]
```

Comment:
Does the limited length of these lists imply a limit on the complexity of the types we can have, or is this just a cosmetic thing?

Reply:
> Does the limited length of these lists imply a limit on the complexity of the types we can have

Yep, but if you add a new builtin with 10 type variables and get a type error because of that, you can always extend the list of names. I just thought it would be unlikely we'd need a builtin with more than 9 type variables.